File: src/contests/cg/puzzle/go/sum-of-divisors.go
// 🎮 CodinGame Puzzle - sum-of-divisors
// https://www.codingame.com/training/medium/sum-of-divisors

package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	// d appears as a divisor in floor(n/d) of the numbers 1..n, so the total
	// sum of divisors is sum over d of d * floor(n/d).
	var n int64
	fmt.Fscan(reader, &n)
	var total int64 = 0
	for d := int64(1); d <= n; d++ {
		total += d * (n / d)
	}
	fmt.Println(total)
}

File: src/contests/cg/puzzle/perl/sum-of-divisors.pl
# 🎮 CodinGame Puzzle - sum-of-divisors
# https://www.codingame.com/training/medium/sum-of-divisors

# d appears as a divisor in floor(n/d) of the numbers 1..n, so the total sum
# of divisors is sum over d of d * floor(n/d).
my $n = <STDIN>;
chomp $n;
my $total = 0;
for my $d (1 .. $n) {
    $total += $d * int($n / $d);
}
print "$total\n";
